src/mame/video/hng64_sprite.hxx

```diff
@@ -50,25 +50,57 @@ void hng64_state::draw_sprites(screen_device &screen, bitmap_rgb32 &bitmap, cons
 	int spriteoffsx = (m_spriteregs[1]>>0)&0xffff;
 	int spriteoffsy = (m_spriteregs[1]>>16)&0xffff;
 
+	// This flips between ingame and other screens for roadedge, where the sprites which are filtered definitely needs to change and the game explicitly swaps the values in the sprite list at the same time.
+	// m_spriteregs[2] could also play a part as it also flips between 0x00000000 and 0x000fffff at the same time
+	// Samsho games also set the upper 3 bits which could be related, samsho games still have some unwanted sprites (but also use the other 'sprite clear' mechanism)
+	// Could also be draw order related, check if it inverts the z value?
+	bool zsort = !(m_spriteregs[0] & 0x01000000);
+
 #if 0
 	for (int iii = 0; iii < 0x0f; iii++)
 		osd_printf_debug("%.8x ", m_videoregs[iii]);
 	osd_printf_debug("\n");
 #endif
 
-	while( source<finish )
+	// start with empty list
+	m_spritelist.clear();
+
+	while(source < finish)
+	{
+		if (source[4]&0x04000000) // disable bit, ss64 rankings ?
+		{
+			source += 8;
+		}
+		else
+		{
+			if ((!zsort && (source[2]&0x07ff0000) != 0x07ff0000) || (zsort && (source[2]&0x07ff0000) != 0))
+			{
+				m_spritelist.emplace_back((source[2]&0x7ff0000)>>16, source);
+				if (source[2]&0x00000100) // inline chain mode
+					source += 8 * (1 + (source[2]&0x0000000f)) * (1 + ((source[2]&0x000000f0)>>4));
+				else
+					source += 8;
+			}
+			else
+				source += 8;
+		}
+	}
+
+	if (zsort)
+		std::sort(m_spritelist.begin(), m_spritelist.end(), std::greater<>());
```

Comment:
This is sorting the pair lexically, i.e. by the key first and then by the pointer.  Is that what you actually want?  I would’ve expected you’d want to use `stable_sort` and only compare the key.

Reply:
At the moment, I'm not sure if their is any precedence for sprites with the same priority key, so it probably wouldn't matter.  If sorting by just the key would be faster, that would be the preferable option.

Reply:
Sorting by key only is faster – you can do it by using a lambda for the comparator, something like `[] (auto const &a, auto const &b) { return a.first < b.first; }` would work.